src/createAnimatedComponent.js

```diff
@@ -229,14 +229,25 @@ export default function createAnimatedComponent(Component) {
         : [this.props.style];
       styles = flattenArray(styles);
       const viewTag = findNodeHandle(this);
+      const viewName = this._obtainViewName();
       styles.forEach((style) => {
-        if (style && style.viewTag !== undefined) {
-          style.viewTag.value = viewTag;
+        if (style) {
+          if (style.viewTag) {
```

Comment:
it looks like we need both viewTag and viewName all of the time. It seem it'd make more sense to just have a single value with both these items. We'd avoid additional checks here and in other places + it'd be a bit more intuitive IMO:

For example:
```
style.viewDescriptor.value = { viewTag, viewName }
```

Reply:
fixed b3f674c646c6d7972e8a0ed50226c8028433ac16